File: .sling/libs/web5/eqela/jk-data/0.32.0/jk.uuid.js
let JkUuidUUIDGenerator = function() {};

JkUuidUUIDGenerator.prototype._t = { "JkUuidUUIDGenerator" : true };
JkUuidUUIDGenerator.prototype._tobj = JkUuidUUIDGenerator;

JkUuidUUIDGenerator.NEW = function() {
	var v = new JkUuidUUIDGenerator;
	return v.CTOR_JkUuidUUIDGenerator();
};

JkUuidUUIDGenerator.prototype.CTOR_JkUuidUUIDGenerator = function() {
	return this;
};

JkUuidUUIDGenerator.generate = function() {
	var random = JkRandomRandom.getSharedInstance();
	var bytes = new ArrayBuffer(16);
	for(var n = 0 ; n < 16 ; n++) {
		bytes[n] = ~(~random.nextInt2(0xff));
	}
	bytes[6] = 0x40 | bytes[6] & 0xf;
	bytes[8] = 0x80 | bytes[8] & 0x3f;
	var hex = JkLangString.forBufferHex(bytes);
	return (JkLangString.safeString((JkLangString.getSubString(hex, 0, 8)))) + ("-") + (JkLangString.safeString((JkLangString.getSubString(hex, 8, 4)))) + ("-") + (JkLangString.safeString((JkLangString.getSubString(hex, 12, 4)))) + ("-") + (JkLangString.safeString((JkLangString.getSubString(hex, 16, 4)))) + ("-") + (JkLangString.safeString((JkLangString.getSubString(hex, 20, 12))));
};

JkUuidUUIDGenerator.forString = function(value) {
	console.log("[jk.uuid.UUIDGenerator.forString] (UUIDGenerator.sling:96:3): Not implemented.");
	return null;
};

JkUuidUUIDGenerator.IS_INSTANCE = function(o) {
	return o != null && o._t != null && o._t["JkUuidUUIDGenerator"] === true;
};

File: .sling/libs/sushi/eqela/jk-data/0.27.0/jk.uuid.lua
jk = jk or {}

jk.uuid = jk.uuid or {}

jk.uuid.UUIDGenerator = {}
jk.uuid.UUIDGenerator.__index = jk.uuid.UUIDGenerator
_vm:set_metatable(jk.uuid.UUIDGenerator, {})

function jk.uuid.UUIDGenerator._create()
	local v = _vm:set_metatable({}, jk.uuid.UUIDGenerator)
	return v
end

function jk.uuid.UUIDGenerator:_init()
	self._isClassInstance = true
	self._qualifiedClassName = self._qualifiedClassName or 'jk.uuid.UUIDGenerator'
	self['_isType.jk.uuid.UUIDGenerator'] = true
end

function jk.uuid.UUIDGenerator:_construct0()
	jk.uuid.UUIDGenerator._init(self)
	return self
end

function jk.uuid.UUIDGenerator:generate()
	local random = _g.jk.random.Random:getSharedInstance()
	local bytes = _util:allocate_buffer(16)
	do
		local n = 0
		while n < 16 do
			bytes[n + 1] = random:nextInt2(255)
			do n = n + 1 end
		end
	end
	bytes[6 + 1] = _vm:bitwise_or(64, _vm:bitwise_and(bytes[6 + 1], 15))
	bytes[8 + 1] = _vm:bitwise_or(128, _vm:bitwise_and(bytes[8 + 1], 63))
	do
		local hex = _g.jk.lang.String:forBufferHex(bytes)
		do return _g.jk.lang.String:safeString(_g.jk.lang.String:getSubString(hex, 0, 8)) .. "-" .. _g.jk.lang.String:safeString(_g.jk.lang.String:getSubString(hex, 8, 4)) .. "-" .. _g.jk.lang.String:safeString(_g.jk.lang.String:getSubString(hex, 12, 4)) .. "-" .. _g.jk.lang.String:safeString(_g.jk.lang.String:getSubString(hex, 16, 4)) .. "-" .. _g.jk.lang.String:safeString(_g.jk.lang.String:getSubString(hex, 20, 12)) end
	end
end

function jk.uuid.UUIDGenerator:forString(value)
	_io:write_to_stdout("[jk.uuid.UUIDGenerator.forString] (UUIDGenerator.sling:97:3): Not implemented." .. "\n")
	do return nil end
end
